src/Magnum/SceneGraph/OctreeDrawableGroup.hpp

```diff
@@ -0,0 +1,113 @@
+#ifndef Magnum_SceneGraph_OctreeDrawableGroup_hpp
+#define Magnum_SceneGraph_OctreeDrawableGroup_hpp
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2015 Andrea Capobianco <andrea.c.2205@gmail.com>
+    Copyright © 2015 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+/** @file
+ * @brief @ref compilation-speedup-hpp "Template implementation" for @ref OctreeDrawableGroup.h
+ */
+
+#include <Magnum/Shapes/AbstractShape.h>
+#include <Magnum/Shapes/ShapeGroup.h>
+
+#include <Magnum/Octree/Octree.hpp>
+
+#include "Magnum/SceneGraph/OctreeDrawableGroup.h"
+
+namespace Magnum { namespace SceneGraph {
+
+using namespace Shapes;
+
+template<class T> OctreeDrawableGroup<T>& OctreeDrawableGroup<T>::add(Drawable<3, T>& drawable, const Range3D boundingBox) {
+    _drawables.push_back(&drawable);
+    _boundingBoxes.push_back(boundingBox);
+
+    if(_octree != nullptr) {
+        /* Destroy previous octree build */
+        delete _octree;
+    }
```

Comment:
Can the instance be not heap-allocated and reused somehow? Or at least used via `std::unique_ptr` and not with manual memory management? This will cause cascading destructor calls and lots of (de)allocations every time.